packages/@aws-cdk/aws-lambda-event-sources/lib/kafka.ts

```diff
@@ -141,6 +157,8 @@ export class SelfManagedKafkaEventSource extends StreamEventSource {
       if (!props.vpcSubnets) {
         throw new Error('vpcSubnets must be set when providing vpc');
       }
+    } else if (!props.secret) {
+      throw new Error('secret must be set if Kafka brokers accessed over Internet');
```

Comment:
Thanks for adding. Don't we need the same for managed kafka as well?

Reply:
From https://www.amazonaws.cn/en/msk/faqs/,

>  Q: Does Amazon MSK run in an Amazon VPC?
> Yes, Amazon MSK always runs within an Amazon VPC managed by the Amazon MSK service.

Reply:
I'm not sure if this means `secret` is only valid for self-managed Kafka or not, so I left it.

Reply:
Let's leave this as is for now. We can come back to it later when we have more information.